### PR TITLE
grafana_plugin: update documentation

### PR DESCRIPTION
**Pull request by theo-o**
making grammatical fixes & overall clarifications

##### SUMMARY

Updates module documentation to clarify some elements & fix spelling errors.

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME
lib/ansible/modules/monitoring/grafana_plugin.py

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by theo-o** [comment]
cc @seuf

**Review by resmo — APPROVED** [review]
(checked botmeta for author handle rename, not changes needed. )

LGTM

File: lib/ansible/modules/monitoring/grafana_plugin.py [hunk 1]
```diff
@@ -16,11 +16,12 @@
 ---
 module: grafana_plugin
 author:
-  - Thierry Sallé (@tsalle)
+  - Thierry Sallé (@seuf)
 version_added: "2.5"
 short_description: Manage Grafana plugins via grafana-cli
 description:
   - Install and remove Grafana plugins.
+  - See L(https://grafana.com/docs/plugins/installation/) for upstream documentation.
```
Comment (acozine) [inline lib/ansible/modules/monitoring/grafana_plugin.py]: Use `U(https://link)` syntax for URL links like this. The `L(...)` syntax requires two elements, `L(link_text,https://link)`. See https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#linking-within-module-documentation for more details.

File: lib/ansible/modules/monitoring/grafana_plugin.py [hunk 2]
```diff
@@ -29,28 +30,31 @@
...
+      - If omitted, defaults to C(/var/lib/grafana/plugins).
   grafana_repo:
     description:
-      - Grafana repository. If not set, gafana-cli will use the default value C(https://grafana.net/api/plugins).
+      - URL to the Grafana plugin repository.
+      - "If omitted, grafana-cli will use the default value: L(https://grafana.com/api/plugins)."
   grafana_plugin_url:
     description:
-      - Custom Grafana plugin URL.
+      - Full URL to the plugin zip file instead of downloading the file from L(https://grafana.com/api/plugins).
```
Comment (acozine) [inline lib/ansible/modules/monitoring/grafana_plugin.py]: Same thing here.

File: lib/ansible/modules/monitoring/grafana_plugin.py [hunk 3]
```diff
@@ -29,28 +30,31 @@
...
       - Requires grafana 4.6.x or later.
   state:
     description:
-      - Status of the Grafana plugin.
-      - If latest is set, the version parameter will be ignored.
-    choices: [ absent, present ]
+      - Defines whether the plugin should be installed.
```
Comment (acozine) [inline lib/ansible/modules/monitoring/grafana_plugin.py]:
```suggestion
      - Whether the plugin should be installed.
```

File: lib/ansible/modules/monitoring/grafana_plugin.py [hunk 4]
```diff
@@ -29,28 +30,31 @@
...
+      - "If omitted, grafana-cli will use the default value: L(https://grafana.com/api/plugins)."
   grafana_plugin_url:
     description:
-      - Custom Grafana plugin URL.
+      - Full URL to the plugin zip file instead of downloading the file from L(https://grafana.com/api/plugins).
       - Requires grafana 4.6.x or later.
   state:
     description:
-      - Status of the Grafana plugin.
-      - If latest is set, the version parameter will be ignored.
```
Comment (acozine) [inline lib/ansible/modules/monitoring/grafana_plugin.py:46]: What's your thinking about removing this note?

File: lib/ansible/modules/monitoring/grafana_plugin.py [hunk 5]
```diff
@@ -60,9 +64,9 @@
 RETURN = '''
 ---
 version:
-  description: version of the installed / removed plugin.
+  description: version of the installed/removed plugin.
```
Comment (acozine) [inline lib/ansible/modules/monitoring/grafana_plugin.py]:
```suggestion
  description: version of the installed/removed/updated plugin.
```

**Review by acozine — COMMENTED** [review]
Thanks for updating the grafana documentation! I made a few suggestions to make the documentation even better.

File: lib/ansible/modules/monitoring/grafana_plugin.py (hunk 4 above)
Comment (theo-o) [reply lib/ansible/modules/monitoring/grafana_plugin.py:46]: `latest` isn't a choice for state, so the statement is incorrect.

**Comment by acozine** [comment]
Thanks @theo-o!

**Comment by samccann** [comment]
Thanks @theo-o for the Ansible documentation fix, and thanks @resmo for  the review comments!

**Pull request merged**